### PR TITLE
[PI-120] temp_apigee_instance

### PR DESCRIPTION
**Pull request by Rohoolio**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,6 +41,16 @@ Do `make build` every time you would like to pick up and install new local/proje
 
 The first time it will also set up your pre-commit hooks.
 
+### Proxygen - WIP
+
+if you wish to deploy proxies to Apigee then there is some additional setup outside of the above due to proxygen not being an ASDF plugin
+
+`pip install proxygen-cli`
```
Comment (jaklinger) [inline README.md:48]: add this to `pyproject.toml` with `poetry proxygen-cli --group dev`

File: src/api/readProduct/index.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+from event.environment import BaseEnvironment
```
Comment (jaklinger) [inline src/api/readProduct/index.py:1]: why did these get added?

File: scripts/proxygen/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+PROXYGEN_COMMAND="$1"
+WORKSPACE="$2"
+ACCOUNT_WIDE="$3"
+
+sh.proxygen()
```
Comment (jaklinger) [inline scripts/proxygen/deploy.sh:7]: ??

File: scripts/infrastructure/swagger/merge.sh [hunk 4]
```diff
@@ -78,7 +94,8 @@ fi
 # * 4XX codes
 if [[ ${MERGE_AWS} == "1" ]]; then
     cat ${_CLEANED_SWAGGER_FILE} |
-        yq 'del(.. | select(has("4XX")).4XX)' > ${AWS_SWAGGER_FILE}
+        yq 'del(.x-*)' |
+        yq 'del(.. | select(has("4XX")).4XX)' > ${AWS_SWAGGER_FILE} |
```
Comment (jaklinger) [inline scripts/infrastructure/swagger/merge.sh:98]: this'll break

File: scripts/proxygen/deploy.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import fire
+
+
+def deploy_proxy(env: str, app_alias="dev"):
+
+    return "Not implemented yet"
```
Comment (jaklinger) [inline scripts/proxygen/deploy.py:6]:
```suggestion
    raise NotImplementedError
```

File: infrastructure/swagger/04_apim.yaml [hunk 6]
```diff
@@ -1,17 +1,15 @@
...
-      grants:
-        nhs-login-p5: []
     - title: Application Restricted
       grants:
         app-level0: []
   target: # Only one target (omit for NO backend)
     type: external
     healthcheck: /_status
-    url: https://example.external.apikey.secured.backend.com
+    url: https://i38bu8egci.execute-api.eu-west-2.amazonaws.com/production #for now this is the dev gateway
```
Comment (jaklinger) [inline infrastructure/swagger/04_apim.yaml:11]: can use templating here?

File: .github/workflows/_deploy.yml [hunk 7]
```diff
@@ -24,7 +24,7 @@ env:
   WORKSPACE: ${{ inputs.workspace }}
   CACHE_NAME: ${{ inputs.workspace }}-cache
   PARAMETER_DEPLOY: ${{ inputs.parameter-deploy }}
-  ACCOUNT_WIDE: ${{ inputs.account-wide }}
+  ACCOUNT_WIDE: ${{ inputs.account-wide  }}
```
Comment (jameslinnell) [inline .github/workflows/_deploy.yml:27]: extra space. Not sure if this has an affect?

Comment (Rohoolio) [reply .github/workflows/_deploy.yml:27]: Not sure how that happened - will remove

File: scripts/proxygen/deploy.py (hunk 5 above)
Comment (Rohoolio) [reply scripts/proxygen/deploy.py:6]: Yeah ill delete the extra proxygen bits - didnt finish it

File: scripts/proxygen/deploy.sh (hunk 3 above)
Comment (Rohoolio) [reply scripts/proxygen/deploy.sh:7]: Yeah ill delete the extra proxygen bits - didnt finish it

File: src/api/readProduct/index.py (hunk 2 above)
Comment (Rohoolio) [reply src/api/readProduct/index.py:1]: Think im supposed to remove all this stuff so i ill do that

File: README.md (hunk 1 above)
Comment (Rohoolio) [reply README.md:48]: Cant do this unfortunately, its not a recognised package for poetry - only way ive managed to install it is with the above

File: scripts/infrastructure/swagger/merge.sh (hunk 4 above)
Comment (Rohoolio) [reply scripts/infrastructure/swagger/merge.sh:98]: Are you sure? I can create a valid swagger file from this when I run the merge, unless we want to delete more x- sections that arent on the top level?

File: infrastructure/terraform/per_account/dev/parameters/main.tf [hunk 8]
```diff
@@ -30,3 +30,7 @@ JSON
 resource "aws_secretsmanager_secret" "rowans-test-secret-account" {
```
Comment (jameslinnell) [inline infrastructure/terraform/per_account/dev/parameters/main.tf:30]: Can we delete this one now?

**Review by jameslinnell — APPROVED** [review]

File: README.md [hunk 9]
```diff
@@ -41,6 +41,14 @@ Do `make build` every time you would like to pick up and install new local/proje
 
 The first time it will also set up your pre-commit hooks.
 
+### Proxygen - WIP
+
+If you wish to deploy proxies to Apigee then we use the proxygen-cli which is isntalled as part of the project
```
Comment (jaklinger) [inline README.md:46]:
```suggestion
If you wish to deploy proxies to Apigee then we use the proxygen-cli which is installed as part of the project
```

**Review by jaklinger — APPROVED** [review]
just a type in the readme, otherwise lgtm

**Pull request merged**